render_files.py

```diff
@@ -154,18 +150,22 @@ def pull_secret(config_dir):
         return yaml.safe_load(yaml_file)['pullSecret']
 
 
-def set_pull_secret(config_dir):
-    with open('/root/.docker/config.json', 'w+') as config_file:
-        config_file.write(pull_secret(config_dir))
+def set_pull_secret(work_dir, config_dir):
+    registry_file_path = os.path.join(work_dir, REGISTRY_CONFIG)
+    with open(registry_file_path, 'w+') as registry_file:
+        registry_file.write(pull_secret(config_dir))
+    return registry_file_path
 
 
-# def prepare_generation_data(work_dir, config_dir, install_config, openshift_release_image):
-def prepare_generation_data(config_dir, install_config):
+def prepare_generation_data(work_dir, config_dir, install_config, openshift_release_image):
+    # set instal-config.yaml
     prepare_install_config(config_dir, install_config)
-    # [TODO] - part of 4.6 , must be solved as part of MGMT-1816
-    # set_pull_secret(config_dir)
-    # [TODO] - remove comment after fixing subsystem
-    # oc_utils.extract_baremetal_installer(work_dir, openshift_release_image)
+    # set pull secret in a file
+    registry_config_file = set_pull_secret(work_dir, config_dir)
+    # extract openshift-installer
```

Comment:
same

Reply:
i think better to comment on the function itself if it is complicated

Reply:
in any case, once we move to 4.6, we will move the code to Golang in assisted-service... The same applies for comment above